core/src/main/scala/kafka/server/AdminManager.scala

```diff
@@ -73,23 +80,29 @@ class AdminManager(val config: KafkaConfig,
             && !arguments.replicasAssignments.isEmpty)
             throw new InvalidRequestException("Both numPartitions or replicationFactor and replicasAssignments were set. " +
               "Both cannot be used at the same time.")
-          else if (!arguments.replicasAssignments.isEmpty) {
-            // Note: we don't check that replicaAssignment doesn't contain unknown brokers - unlike in add-partitions case,
-            // this follows the existing logic in TopicCommand
-            arguments.replicasAssignments.asScala.map { case (partitionId, replicas) =>
-              (partitionId.intValue, replicas.asScala.map(_.intValue))
+          else {
+            createTopicPolicy.foreach(_.validate(new CreateTopicPolicy.TopicDetails(topic, arguments.numPartitions,
+              arguments.replicationFactor, arguments.replicasAssignments, arguments.configs)))
+
+            if (!arguments.replicasAssignments.isEmpty) {
+              // Note: we don't check that replicaAssignment doesn't contain unknown brokers - unlike in add-partitions case,
+              // this follows the existing logic in TopicCommand
+              arguments.replicasAssignments.asScala.map { case (partitionId, replicas) =>
+                (partitionId.intValue, replicas.asScala.map(_.intValue))
+              }
+            } else {
+              AdminUtils.assignReplicasToBrokers(brokers, arguments.numPartitions, arguments.replicationFactor)
             }
-          } else {
-            AdminUtils.assignReplicasToBrokers(brokers, arguments.numPartitions, arguments.replicationFactor)
           }
         }
         trace(s"Assignments for topic $topic are $assignments ")
-        AdminUtils.createOrUpdateTopicPartitionAssignmentPathInZK(zkUtils, topic, assignments, configs, update = false)
-        CreateTopicMetadata(topic, assignments, Errors.NONE)
+        AdminUtils.createOrUpdateTopicPartitionAssignmentPathInZK(zkUtils, topic, assignments, configs,
+          update = false, validateOnly = validateOnly)
+        CreateTopicMetadata(topic, assignments, new CreateTopicsResponse.Error(Errors.NONE, null))
       } catch {
         case e: Throwable =>
-          warn(s"Error processing create topic request for topic $topic with arguments $arguments", e)
-          CreateTopicMetadata(topic, Map(), Errors.forException(e))
+          info(s"Error processing create topic request for topic $topic with arguments $arguments", e)
```

Comment:
I guess its a matter of taste :) IMO, it requires attention since it can indicate an unhappy user - maybe reaching out to the user to discuss why he's trying to do what he does? or to explain why he can't? Maybe not all admins are as caring :)

If you are not convinced, leave it as is. We can figure out if admins complain later :)